platform/tests/minion_manager/aws/aws_bid_advisor_test.py

```diff
@@ -0,0 +1,128 @@
+"""The file has unit tests for the AWSBidAdvisor."""
+
+import unittest
+
+from ax.platform.minion_manager.cloud_provider.aws.aws_bid_advisor import AWSBidAdvisor
+
+REFRESH_INTERVAL = 10
+REGION = 'us-west-2'
+
+
+class AWSBidAdvisorTest(unittest.TestCase):
+    """
+    Tests for AWSBidAdvisor.
+    """
+    def test_ba_lifecycle(self):
+        """
+        Tests that the AWSBidVisor starts threads and stops them correctly.
+        """
+        bidadv = AWSBidAdvisor(REFRESH_INTERVAL, REFRESH_INTERVAL, REGION)
+        assert len(bidadv.all_bid_advisor_threads) == 0
+        bidadv.run()
+        assert len(bidadv.all_bid_advisor_threads) == 2
+        bidadv.shutdown()
+        assert len(bidadv.all_bid_advisor_threads) == 0
+
+    def test_ba_on_demand_pricing(self):
+        """
+        Tests that the AWSBidVisor correctly gets the on-demand pricing.
+        """
+        bidadv = AWSBidAdvisor(REFRESH_INTERVAL, REFRESH_INTERVAL, REGION)
+        assert len(bidadv.on_demand_price_dict) == 0
+        updater = bidadv.OnDemandUpdater(bidadv)
+        updater.get_on_demand_pricing()
+        assert len(bidadv.on_demand_price_dict) > 0
```

Comment:
len > 0 might be a weak assertion, but cannot think of any better way to test it.